stories/core/Table.stories.tsx

```diff
@@ -718,3 +759,119 @@ export const NoData: Story<TableProps> = ({ columns, data, ...rest }) => {
 NoData.args = {
   data: [],
 };
+
+export const ControlledState: Story<TableProps> = (args) => {
+  const { columns, data, ...rest } = args;
+
+  const [selectedRows, setSelectedRows] = useState<Record<string, boolean>>({});
+
+  const onSelect = useCallback((rows, state) => {
+    action(
+      `Selected rows: ${JSON.stringify(rows)}, Table state: ${JSON.stringify(
+        state,
+      )}`,
+    )();
+  }, []);
+
+  const tableColumns = useMemo(
+    () => [
+      {
+        Header: 'Table',
+        columns: [
+          {
+            id: 'name',
+            Header: 'Name',
+            accessor: 'name',
+          },
+          {
+            id: 'description',
+            Header: 'Description',
+            accessor: 'description',
+          },
+        ],
+      },
+    ],
+    [],
+  );
+
+  const tableData = useMemo(
+    () => [
+      { name: 'Name1', description: 'Description1' },
+      { name: 'Name2', description: 'Description2' },
+      { name: 'Name3', description: 'Description3' },
+    ],
+    [],
+  );
+
+  const controlledState = useCallback(
+    (state) => {
+      return {
+        ...state,
+        selectedRowIds: { ...selectedRows },
+      };
+    },
+    [selectedRows],
+  );
+
+  // When using `useControlledState` we are fully responsible for the state part we are modifying.
+  // Therefore we want to keep our outside state (`selectedRows`) in sync with inside table state (`state.selectedRowIds`).
+  const tableStateReducer = (
+    newState: TableState,
+    action: ActionType,
+  ): TableState => {
+    switch (action.type) {
+      case actions.toggleRowSelected: {
```

Comment:
Need to also add `toggleAllRowsSelected` because that's broken right now

Reply:
Handling.